File: DotNetAssignments/assign5/loanlib/Loan.cs
namespace loan;

public abstract class Loan
{
    public double Principle { get; set; }
    public int Period { get; set; }

    public abstract getRate(double rate);

    public getEMI(double amount){
        // emi = p * (1 + r * n / 100) / (12 * n)
    }
}

File: Java_master/JavaAssignment/akash/the loan player/loan/Loan.java
package loan;

public abstract class Loan {
    private double principle;
    private double duration;

    public Loan(double principle, double duration) {
        this.principle = principle;
        this.duration = duration;
    }

    public void setPrinciple(double amount) {
        this.principle = amount;
    }

    public double getPrinciple() {
        return principle;
    }

    public void setDuration(double months) {
        this.duration = months;
    }

    public double getDuration() {
        return duration;
    }

    public abstract double getRate();

    public double getEMI() {
        return getPrinciple() * (1 + getRate() * getDuration() / 100)/(12*getDuration());
    } 

}